### DISCUSSION TITLE
how to get a solid structure

### DISCUSSION THREAD

**Discussion by utopiankk** [PMEAL/porespy#901] [Q&A]
Hello Everyone! 
I've used PoreSpy to generate a fibrous porous structure, but  only get images. Could you please guide me on how to export it as a 3D solid structure?

**Comment by jgostick** [comment]
You'll have to elaborate on what you mean by solid structure?  The images are binary images with `True` indicating pore space and `False` for solid.  So if you mean solid as in an image of the solid then do `im = ~im` (i.e. just invert the image).  If you mean something you can view in 3D, then export the image using `ps.io.to_vtk()` and open in paraview.  Other than that, I'm not sure what you're asking.

**Reply by utopiankk** [reply]
Thank you very much for your reply.
 I used porespy to generate a fiber porous structure, which is a binary space matrix, and I output it as a 3D tif slice, which can be rendered in imageJ or other software to generate a 3D solid structure. I want to output this 3D solid structure as a general 3D solid model for finite element analysis. For example, the format is.stp. Anyway, what I want to do is output the binary spatial structure generated by the code into a common format for the 3D model.
Thank you, Professor.

**Reply by jgostick** [reply]
converting a binary image to a mesh is a separate challenge outside of what porespy/openpnm can do.  I have had success with nanomesh in the past, though not sure how well it scales to large images.  Good luck!

**Reply by utopiankk** [reply]
Thank you Professor for your kind reply！
I am trying this operation, but it does require some hard work and luck.

**Reply by utopiankk** [reply]
Hello, Professor! Thank you very much for your kind reply before, and I still have a small question to consult.

I generate the fiber porous structure through ps.generators.cylinders in porespy. I wonder if the code can export the coordinates of the two endpoints of the fiber.

**Reply by jgostick** [reply]
No, I'm afraid not.  The fibers are generated by choosing a random point and a random angle, then extending it out in both directions.  This is done using the `line_segment` function, so maybe you can use that to implement something that will work for you.

**Reply by utopiankk** [reply]
Ok！Thanks again professor, I'll give it a try.
